### PR TITLE
Core: Switch tests to JUnit5 in avro, data.avro pakages

### PR DESCRIPTION
**Pull request by skytin1004**
This pull request is related to the issue "Move JUnit4 tests to JUnit5 https://github.com/apache/iceberg/issues/7160".
I've switched the all files in Core: avro, data.avro pakages to JUnit5.

### CODE REVIEW

**Comment by skytin1004** [comment]
@nastra Could you please review this PR when you have a moment?
Thank you in advance.

File: core/src/test/java/org/apache/iceberg/avro/TestAvroNameMapping.java [hunk 1]
```diff
@@ -369,15 +377,15 @@ protected Record writeAndRead(
...
+    Assertions.assertThat(projectedWithNameMapping).isEqualTo(record);
     return record;
   }
 
   private Record writeAndRead(
       Schema writeSchema, Schema readSchema, Record record, NameMapping nameMapping)
       throws IOException {
 
-    File file = temp.newFile();
+    File file = new File(temp, "test.avro");
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestAvroNameMapping.java:388]: should this be handled similarly to https://github.com/apache/iceberg/pull/8056/files#diff-d4debd1e3696fc5a7684ef1cf178c7d100aedfad26ffbca0560038eb98e90487R98?

File: core/src/test/java/org/apache/iceberg/avro/TestAvroReadProjection.java [hunk 2]
```diff
@@ -30,16 +30,15 @@
...
+import org.junit.jupiter.api.Test;
 
 public class TestAvroReadProjection extends TestReadProjection {
   @Override
   protected GenericData.Record writeAndRead(
       String desc, Schema writeSchema, Schema readSchema, GenericData.Record record)
       throws IOException {
-    File file = temp.newFile(desc + ".avro");
-    file.delete();
+    File file = new File(temp, desc + ".avro");
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestAvroReadProjection.java:41]: same question w.r.t new file handling

**Comment by skytin1004** [comment]
@nastra 
Thank you for the review, I've modified the code based on your review.
 I created a method "createAvroTempFile" in the "AvroWritingTestUtil", referring to the link you attached. 
Please review it when you have time. If there are any problems with the method, I will be happy to modify it.

File: core/src/test/java/org/apache/iceberg/avro/TestAvroNameMapping.java [hunk 3]
```diff
@@ -369,15 +378,15 @@ protected Record writeAndRead(
...
+    Assertions.assertThat(projectedWithNameMapping).isEqualTo(record);
     return record;
   }
 
   private Record writeAndRead(
       Schema writeSchema, Schema readSchema, Record record, NameMapping nameMapping)
       throws IOException {
 
-    File file = temp.newFile();
+    File file = createAvroTempFile(temp);
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestAvroNameMapping.java:389]: why not just `temp.resolve("avro").toFile()`?

File: core/src/test/java/org/apache/iceberg/avro/AvroWritingTestUtils.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.UUID;
+import org.apache.iceberg.FileFormat;
+
+public class AvroWritingTestUtils {
+
+  private AvroWritingTestUtils() {}
+
+  static File createAvroTempFile(Path temp) throws IOException {
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/AvroWritingTestUtils.java:31]: I don't think we need this class. See my other comments

File: core/src/test/java/org/apache/iceberg/avro/TestReadProjection.java [hunk 5]
```diff
@@ -329,50 +346,57 @@ public void testMapOfStructsProjection() throws IOException {
...
     projected = writeAndRead("lat_only", writeSchema, writeSchema.select("locations.lat"), record);
     assertEmptyAvroField(projected, "id");
     Map<String, ?> locations = toStringMap((Map) projected.get("locations"));
-    Assert.assertNotNull("Should project locations map", locations);
-    Assert.assertEquals(
-        "Should contain L1 and L2", Sets.newHashSet("L1", "L2"), locations.keySet());
+    Assertions.assertThat(locations).as("Should project locations map").isNotNull();
+    Assertions.assertThat(locations.keySet())
+        .as("Should contain L1 and L2")
+        .isEqualTo(Sets.newHashSet("L1", "L2"));
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestReadProjection.java:366]: .containsExactly("L1", "L2")

**Comment by skytin1004** [comment]
Thank you for the detailed review, I've modified the code based on your review. Could you please review this PR when you have a moment?

**Comment by skytin1004** [comment]
@nastra Thank you for the detailed review, I've modified the code based on your review. Could you please review this PR when you have a moment?

**Review by nastra — APPROVED** [review]
thanks @skytin1004

**Pull request merged**